framework/components/ecl-carousels/carousels.js

```diff
@@ -0,0 +1,175 @@
+import {
+  addClass,
+  removeClass,
+  queryAll,
+} from '@ec-europa/ecl-base/helpers/dom';
+
+import debounce from 'lodash.debounce';
+
+/**
+ * @param {object} options Object containing configuration overrides
+ */
+export const carousels = ({ selectorId: selectorId = 'ecl-carousel' } = {}) => {
+  // SUPPORTS
+  if (!('querySelector' in document) || !('addEventListener' in window)) {
+    return null;
+  }
+
+  // SETUP
+  let currentSlide = 0;
+  const carousel = document.getElementById(selectorId);
+  const slides = queryAll('.ecl-carousel__item', carousel);
+  const list = carousel.querySelector('.ecl-carousel__list');
+
+  function getListItemWidth() {
+    return carousel.querySelector('.ecl-carousel__item').offsetWidth;
+  }
+
+  function alignCenter() {
+    slides.forEach(slide => {
+      // eslint-disable-next-line
+      slide.style.left = `${(list.offsetWidth - slide.offsetWidth) / 2}px`;
+    });
+  }
+
+  function goToSlide(n) {
+    removeClass(slides[currentSlide], 'ecl-carousel__item--showing');
+    currentSlide = (n + slides.length) % slides.length;
+    addClass(slides[currentSlide], 'ecl-carousel__item--showing');
+  }
+
+  function setOffset() {
+    const itemWidth = getListItemWidth();
+    const tr = `translate3d(${-currentSlide * itemWidth}px, 0, 0)`;
+
+    list.style.MozTransform = tr; /* FF */
+    list.style.msTransform = tr; /* IE (9+) */
+    list.style.OTransform = tr; /* Opera */
+    list.style.WebkitTransform = tr; /* Safari + Chrome */
+    list.style.transform = tr;
+  }
+
+  function announceCurrentSlide() {
+    carousel.querySelector(
+      '.ecl-carousel__meta-slide'
+    ).textContent = `${currentSlide + 1} / ${slides.length}`;
+  }
+
+  function showImageInformation() {
+    // Reset/Hide all.
+    const infoAreas = queryAll('[data-image]');
+    // If anything is visible.
+    if (infoAreas) {
+      // eslint-disable-next-line
+      infoAreas.forEach(area => (area.style.display = "none"));
+    }
+
+    carousel.querySelector(`[data-image="${currentSlide}"]`).style.display =
+      'block';
+  }
+
+  function previousSlide() {
+    goToSlide(currentSlide - 1);
+    alignCenter();
+    setOffset();
+    announceCurrentSlide();
+    showImageInformation();
+  }
+
+  function nextSlide() {
+    goToSlide(currentSlide + 1);
+    alignCenter();
+    setOffset();
+    announceCurrentSlide();
+    showImageInformation();
+  }
+
+  // Attach controls to a carousel.
+  function addCarouselControls() {
+    const navControls = document.createElement('ul');
```

Comment:
It's true it would have been easier just placing 2 buttons independently as many carousels do, though this [part of the best practices](https://www.w3.org/WAI/tutorials/carousels/functionality/#add-navigation-buttons) I implemented.